### PR TITLE
Refactor test seeds config for E2E tests

### PR DESCRIPTION
**Pull request by Pezmc**
Extra coverage needs to be introduced for the deployments screen, which involves lots of additions to this test setup and it was becoming very unwieldy and hard to follow. This PR refactors the test setup to use the factory pattern. It also fixes an issue where team2 was using project template/types/stacks owned by Alice who is on team 1.

Follow-up PR which introduces the config for testing deployments coming shortly.

### CODE REVIEW

**Comment by knolleary** [comment]
Looks good. We should consider making the factory available at a higher level for other test areas that use this pattern. Not a priority for 1.1 though.

**Comment by Pezmc** [comment]
Will look into test failures once other PR is open

**Comment by Pezmc** [comment]
@knolleary I like that idea, a single factory shared across unit/integration tests.
